### ISSUE TITLE
[MNG-4463] Dependency management import should support version ranges

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#5552]
**[Rob ten Hove](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=robth)** opened **[MNG-4463](https://issues.apache.org/jira/browse/MNG-4463?redirect=false)** and commented

Version ranges cannot be used for artifacts with `import` scope. If a version range is used for such an artifact, Maven cannot find it. Looking at the console output shows that it takes the version range as the version, without resolving it:

```
Downloading: http://some-repo/group/artifact/[1.0.0,2.0.0)/artifact-[1.0.0,2.0.0).pom
```

This is the POM snippet:

```xml
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>group</groupId>
      <artifactId>artifact</artifactId>
      <version>[1.0.0,2.0.0)</version>
      <!-- Using e.g. <version>1.1.0</version> does work! -->
      <type>pom</type>
      <scope>import</scope>
    </dependency>
  </dependencies>
</dependencyManagement>
```

---

**Affects:** 2.2.1

**Issue Links:**
- [MNG-6656](https://issues.apache.org/jira/browse/MNG-6656) Introduce base for build/consumer process
 (_**"is part of"**_)

**Remote Links:**
- [Page
](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=67636510)
- [GitHub Pull Request #28
](https://github.com/apache/maven-integration-testing/pull/28)
- [GitHub Pull Request #146
](https://github.com/apache/maven/pull/146)
- [GitHub Pull Request #28
](https://github.com/apache/maven-integration-testing/pull/28)
- [GitHub Pull Request #146
](https://github.com/apache/maven/pull/146)

**Backported to:** [4.0.0-alpha-2](https://github.com/apache/maven/milestone/113?closed=1)

17 votes, 32 watchers

**Comment by jira-importer** [comment]
**[Robert Thornton](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rptmaestro)** commented

Does this fix also allow expressions in the version?  For example:

```xml
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>group</groupId>
      <artifactId>artifact</artifactId>
      <version>${artifact.version}</version>
      <type>pom</type>
      <scope>import</scope>
    </dependency>
  </dependencies>
</dependencyManagement>
```

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Implicitly fixed with build/consumer POM and with [3bcfddafaa85e2ad6279bbffb13d4fd676124c31](https://gitbox.apache.org/repos/asf?p=maven-integration-testing.git;a=commit;h=3bcfddafaa85e2ad6279bbffb13d4fd676124c31).
